src/index.d.ts

```diff
@@ -0,0 +1,25 @@
+// index.d.ts
+declare module "@zerodevx/svelte-toast";
+
+import type { SvelteComponent } from "svelte";
+
+export interface SvelteToastOptions {
+  duration: number;
+  dismissable: boolean;
+  initial: number;
+  progress: number;
+  reversed: boolean;
+  intro: any;
+  theme: { [key: string]: string }
+}
+
+export class SvelteToast extends SvelteComponent {
+  options: SvelteToastOptions;
+}
+
+declare namespace toast {
+  export function push(text: string, options?: SvelteToastOptions): number;
+  export function pop(id: number): void;
+  export function set(id: number, options: SvelteToastOptions): void;
+  export function _opts(options: SvelteToastOptions): SvelteToastOptions;
+}
```

Comment:
Lastly, is it OK to run this file through [standardjs](https://standardjs.com/) for code formatting, or is it an anti-pattern?

Reply:
I can't say I've ever tried. I've stuck to the conventions I see when typescript generates its own definition files, so I'm inclined to say anti-pattern, but that's based purely on what I've inferred.